app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/common/view/BaseFragment.java

```diff
@@ -63,23 +61,10 @@
     @Nullable
     private Unbinder viewUnbinder;
 
-    @SuppressWarnings("deprecation")
-    @Override
-    public void onAttach(Activity activity) {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
-            // Perform injection here before M, L (API 22) and below because onAttach(Context)
-            // is not yet available at L.
-            AndroidInjection.inject(this);
-        }
-        super.onAttach(activity);
-    }
-
     @Override
     public void onAttach(Context context) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            // Perform injection here for M (API 23) due to deprecation of onAttach(Activity).
-            AndroidInjection.inject(this);
-        }
+        // This is called even for API levels below 17.
+        AndroidSupportInjection.inject(this);
```

Comment:
This is now only invoked in `onAttach(Context)` because the support `Fragment` provides this method even to devices running API level 22 and below.